src/core/Elsa.Abstractions/Models/Variables.cs

```diff
@@ -44,7 +48,7 @@ public Variable SetVariable(string name, object value)
         public void SetVariables(IEnumerable<KeyValuePair<string, Variable>> variables)
         {
             foreach (var variable in variables)
-                SetVariable(variable.Key, variable.Value);
+                SetVariable(variable.Key, variable.Value.Value);

```

Comment:
I feel like the `SetVariable` method needs to have an overload that takes a Variable object. Then that can redirect the `variable.Value` to `SetVariable` 

That should ensure no one accidentally sets a variable to a variable like this.

Im sure the compiler would like to the more specific type if there were two overloads:
```csharp
public Variable SetVariable(string name, object value){ ... }

public Variable SetVariable(string name, Variable variable)
{
    return SetVariable(name, variable.Value);
}
```

Reply:
or maybe just add an `is Variable` test in the current `SetVariable` method?

Reply:
I think the overload version makes sense. The compiler will pick the right one when a Variable is specified.

Reply:
Nice suggestion @petedavis. I've just pushed an update. Nice to know this won't accidentally happen again with this overload in there.

@sfmskywalker What are your thoughts on unit/integration tests? Do you think they'd provide value in this project? If so, do you have any preferences on frameworks/project structure etc?

Reply:
@jamesdoran Yes, I was thinking not too long ago that we need both kinds of tests. I really like xUnit, but I'm open to suggestions.